app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -243,21 +251,50 @@ class ConversationListViewModel @Inject constructor(
         blockUserDialogState = PreservedState.State(BlockUserDialogState(name, id))
     }
 
-    fun leaveGroup(conversationId: ConversationId) = viewModelScope.launch(dispatchers.io()) {
-        leaveGroupConversation(conversationId = conversationId, selfUserId)
+    fun leaveGroup(leaveGroupState: GroupDialogState) {
+        viewModelScope.launch {
+            requestInProgress = true
+            val response = withContext(dispatcher.io()) {
+                val selfUser = observeSelfUser().first()
+                removeMemberFromConversation(
+                    leaveGroupState.conversationId,
+                    selfUser.id
+                )
+            }
+            when (response) {
+                is RemoveMemberFromConversationUseCase.Result.Failure ->
+                    snackBarState.emit(HomeSnackbarState.LeaveConversationError)
+                RemoveMemberFromConversationUseCase.Result.Success -> {
+                    snackBarState.emit(HomeSnackbarState.LeftConversationSuccess)
+                }
+            }
+            requestInProgress = false
+        }
     }
 
-    private fun List<ConversationDetails>.toConversationItemList(teamId: TeamId?): List<ConversationItem> =
+    fun deleteGroup(groupDialogState: GroupDialogState) {
+        viewModelScope.launch {
+            requestInProgress = true
+            when (withContext(dispatcher.io()) { deleteTeamConversation(groupDialogState.conversationId) }) {
+                is Result.Failure.GenericFailure -> snackBarState.emit(HomeSnackbarState.DeleteConversationGroupError)
+                Result.Failure.NoTeamFailure -> snackBarState.emit(HomeSnackbarState.DeleteConversationGroupError)
```

Comment:
```suggestion
                is Result.Failure.GenericFailure, Result.Failure.NoTeamFailure -> snackBarState.emit(HomeSnackbarState.DeleteConversationGroupError)
```

I think this is the equivalent, is just a minor style thingy, so don't take it into account at least you really think helps in some way =)

Reply:
This needs to be refactored to new approach of handling snackbar messages 😄